af_aisf_jobseeker_matching_sync/models/res_partner.py

```
@@ -0,0 +1,441 @@
+from odoo import models, api, registry
+import odoo
+from uuid import uuid4
+from psycopg2 import IntegrityError, InternalError
+import traceback
+import time
+import logging
+
+_logger = logging.getLogger(__name__)
+
+RASK_SYNC = "RASK SYNC"
+MASK_SYNC = "MASK SYNC"
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def _aisf_sync_jobseeker_matching(
+        self,
+        db_values,
+        process_name,
+        customer_id,
+        social_sec_nr=False,
+        eventid=None,
+        batch=None,
+    ):
+        """Perform a sync of a jobseeker from AIS-F.
+        :param customer_id: Customer ID of the jobseeker.
+        :param social_sec_nr: Optional social security number of the jobseeker.
+        :param eventid: The MQ message id for logging purposes.
+        :returns: True if sync succeeded.
+        """
+        log = self.env["af.process.log"]
+        if not batch:
+            log.log_message(process_name, eventid, "SYNC STARTED", objectid=customer_id)
+
+        partner = self.env["res.partner"].search(
+            [("customer_id", "=", customer_id), ("is_jobseeker", "=", True)]
+        )
+        if not partner and social_sec_nr:
+            partner = self.env["res.partner"].search_pnr(social_sec_nr)
+        eventid = eventid or uuid4()
+
+        mask_matching = self.env.ref("af_ipf.ipf_endpoint_mask_matching").sudo()
+        try:
+            if not batch:
+                log.log_message(process_name, eventid, MASK_SYNC, objectid=customer_id)
+            start_time = time.time()
+            res = mask_matching.call(customer_id=int(customer_id))
+            end_time = time.time()
+            time_for_call_to_mask_matching = end_time - start_time
+        except Exception:
```

We want to swallow all errors or are there specific errors we should catch?